MIMETypeRegistry::isPostScriptMIMEType() is no longer a relevant question

aprotyas opened the pull request:
#### 8694a007981ee2d1fc99680947127b6596d9a891
<pre>
MIMETypeRegistry::isPostScriptMIMEType() is no longer a relevant question
<a href="https://bugs.webkit.org/show_bug.cgi?id=273858">https://bugs.webkit.org/show_bug.cgi?id=273858</a>
<a href="https://rdar.apple.com/127706013">rdar://127706013</a>

Reviewed by Tim Horton.

After 278471@main, the only callers of isPostScriptMIMEType() are
Image::create() and CachedImage::updateBufferInternal(), both of which
use the check to convert PostScript data to PDF data. Since we no longer
ship this translation capability, we no longer need to check for the
presence of a PostScript resource.

* Source/WebCore/loader/cache/CachedImage.cpp:
(WebCore::CachedImage::updateBufferInternal):
(WebCore::CachedImage::isPostScriptResource const): Deleted.
* Source/WebCore/loader/cache/CachedImage.h:
* Source/WebCore/platform/MIMETypeRegistry.cpp:
(WebCore::MIMETypeRegistry::isPostScriptMIMEType): Deleted.
* Source/WebCore/platform/MIMETypeRegistry.h:
* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::create):
(WebCore::Image::isPostScriptResource): Deleted.
* Source/WebCore/platform/graphics/Image.h:

Canonical link: <a href="https://commits.webkit.org/278520@main">https://commits.webkit.org/278520@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9c4d525047293d70d4db6b1b82db2b910c11ef8f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50735 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30032 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3052 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53994 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1426 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/53038 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36294 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1076 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41343 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52834 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27678 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43703 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22467 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/25045 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/959 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/9176 "Built successfully") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/44071 "Build is in progress. Recent messages:") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/47026 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1021 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55584 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25837 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/913 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48758 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27094 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43816 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47825 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11135 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27962 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26826 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebCore/loader/cache/CachedImage.cpp:
```diff
@@ -521,21 +516,10 @@ void CachedImage::updateBufferInternal(const FragmentedSharedBuffer& data)
-        return;
-#else
-        // Set the encodedDataStatus to Error so loading this image will be canceled.
-        encodedDataStatus = EncodedDataStatus::Error;
```
You've changed the behavior a bit here, haven't you? Do we still want to cancel the load like we used to?

hortont424 replied:
(just, without an exception on macOS)

shallawa replied:
I think this is fine. I do not think we need to have a special case for postscript images here. 

The call `updateImageData()` below will set the `Image` with the encoded data. If postscript is not supported, it will be treated as a `BitmapImage`. And in this case `ImageDecoder::encodedDataStatus()` will return `EncodedDataStatus::Error`.

hortont424 replied:
Excellent!

aprotyas committed:
MIMETypeRegistry::isPostScriptMIMEType() is no longer a relevant question
https://bugs.webkit.org/show_bug.cgi?id=273858
rdar://127706013

Reviewed by Tim Horton.

After 278471@main, the only callers of isPostScriptMIMEType() are
Image::create() and CachedImage::updateBufferInternal(), both of which
use the check to convert PostScript data to PDF data. Since we no longer
ship this translation capability, we no longer need to check for the
presence of a PostScript resource.

* Source/WebCore/loader/cache/CachedImage.cpp:
(WebCore::CachedImage::updateBufferInternal):
(WebCore::CachedImage::isPostScriptResource const): Deleted.
* Source/WebCore/loader/cache/CachedImage.h:
* Source/WebCore/platform/MIMETypeRegistry.cpp:
(WebCore::MIMETypeRegistry::isPostScriptMIMEType): Deleted.
* Source/WebCore/platform/MIMETypeRegistry.h:
* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::create):
(WebCore::Image::isPostScriptResource): Deleted.
* Source/WebCore/platform/graphics/Image.h:

Canonical link: https://commits.webkit.org/278520@main
